testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ElytronClientDefaultSSLContextProviderTestCase.java

```diff
@@ -0,0 +1,96 @@
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Test that the ElytronClientDefaultSSLContextProvider can be used to configure SSL context for RESTEasy client by calling SSLContext.getDefault()
+ */
+public class ElytronClientDefaultSSLContextProviderTestCase {
+
+    private static final String CONFIG_FILE = "file:./src/test/resources/wildfly-config-default-ssl-context.xml";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final char[] PASSWORD = "password".toCharArray();
+    private static final String ALIAS = "alias";
+    private static File KEYSTORES_DIR = new File("./target/keystores");
+
+    @BeforeClass
+    public static void generateKeystoreAndConfigureProvider() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        if (KEYSTORES_DIR.exists()) {
+            new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
+            KEYSTORES_DIR.delete();
+        }
+        generateClientKeystore();
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+    }
+
+    @Test
+    public void testDefaultSSLContextIsFromElytronClientProvider() {
+        Assert.assertNotNull(Security.getProvider(ElytronClientDefaultSSLContextProvider.class.getSimpleName()));
+        try {
+            ResteasyClientBuilder builder = (ResteasyClientBuilder) ClientBuilder.newBuilder();
+            ResteasyClient resteasyClient = builder.sslContext(SSLContext.getDefault()).build();
+            Assert.assertEquals(resteasyClient.getSslContext().getProvider().getName(), ElytronClientDefaultSSLContextProvider.class.getSimpleName());
+        } catch (NoSuchAlgorithmException e) {
+            Assert.fail("ElytronClientDefaultSSLContextProvider did not provide default SSLContext successfully");
+        }
+    }
+
+    private static void generateClientKeystore() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        if (!KEYSTORES_DIR.exists()) {
+            KEYSTORES_DIR.mkdirs();
+        }
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        KeyStore clientKeyStore = KeyStore.getInstance("JKS");
+        clientKeyStore.load(null, null);
+
+        KeyPair clientKeyPair = keyPairGenerator.generateKeyPair();
+        PrivateKey signingKey = clientKeyPair.getPrivate();
+        PublicKey publicKey = clientKeyPair.getPublic();
+
+        X500Principal issuerDN = new X500Principal("CN=" + ALIAS);
+        X509Certificate clientCertificate = new X509CertificateBuilder()
+                .setIssuerDn(issuerDN)
+                .setSubjectDn(new X500Principal("OU=Elytron"))
+                .setSignatureAlgorithmName("SHA1withRSA")
```

Comment:
May be good to use SHA256withRSA here instead. I think we ran into some testsuite failures recently with SHA1withRSA.

Reply:
> Just realized that the SSLContext won't actually be used in these tests (other than getting the default). Do you have a test case that makes use of the SSLContext?

Thanks for this note, @fjuma 
Initially I thought this could be avoided since the SSL context is tested elsewhere for sure, but after discussing this with @Skyllarr I am now on the same page for having it used for a server call.

Reply:
@fjuma Thanks, cipher changed